datafusion/expr/src/utils.rs

```diff
@@ -42,7 +42,7 @@ use sqlparser::ast::{ExceptSelectItem, ExcludeSelectItem, WildcardAdditionalOpti
 
 ///  The value to which `COUNT(*)` is expanded to in
 ///  `COUNT(<constant>)` expressions
-pub const COUNT_STAR_EXPANSION: ScalarValue = ScalarValue::UInt8(Some(1));
+pub const COUNT_STAR_EXPANSION: ScalarValue = ScalarValue::Int64(Some(1));
```

Comment:
This does however result in `count(1)` being auto-named to `count(*)` now.
```bash
❯ explain select count(1) from hits;
+---------------+---------------------------------------------------+
| plan_type     | plan                                              |
+---------------+---------------------------------------------------+
| logical_plan  | Aggregate: groupBy=[[]], aggr=[[COUNT(Int64(1))]] |
|               |   TableScan: hits projection=[]                   |
| physical_plan | ProjectionExec: expr=[99997497 as COUNT(*)]       |
|               |   PlaceholderRowExec                              |
|               |                                                   |
+---------------+---------------------------------------------------+
2 row(s) fetched.
Elapsed 0.263 seconds.

❯ select count(1) from hits;
+----------+
| COUNT(*) |
+----------+
| 99997497 |
+----------+
1 row(s) fetched.
Elapsed 0.015 seconds.
```
On the other hand that might be fine as it reduces ambiguity.

Reply:
> This does however result in count(1) being auto-named to count(*) now.

FWIW, this is what happens on main with `count(uint8)` (the same thing)

```sql

❯ select count(arrow_cast(1, 'UInt8')) from (values (1));
+----------+
| COUNT(*) |
+----------+
| 1        |
+----------+
1 row(s) fetched.
Elapsed 0.002 seconds.
```

Though admittedly almost no one would actually type `count(arrow_cast(1, 'UInt8'))` so it is likely not a big deal.

> That might be fine as it reduces ambiguity.

I agree having the `COUNT(*)` in the plan actually helps as it makes it clearer when the fast path is being used. 

Let's start with this and if someone else has a different opinion we can make another PR